### PR TITLE
Move to new plugin system.

### PR DESCRIPTION
**Pull request by kleinmann**

### CODE REVIEW

File: Helpers/LicenseHelper.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /**
+     * Add a license key using SwagLicense
+     *
+     * @param string $licenseKey
+     */
+    public function addLicenseKey($licenseKey)
+    {
+        if (method_exists($this->licenseService, 'importLicence')) { // SW 5.5 compatibility
```
Comment (sobyte) [inline Helpers/LicenseHelper.php:30]: Shouldn't we add an alternative or at least a message?

File: Helpers/ProjectHelper.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                $this->modelManager->flush();
+            }
+        }
+
+        $this->modelManager->flush();
+    }
+
+    public function updateCmsSiteGroupNames(array $cmsGroups)
+    {
+        $groupRepo = Shopware()->Models()->getRepository('Shopware\Models\Site\Group');
```
Comment (sobyte) [inline Helpers/ProjectHelper.php]: Please use `Group::class` and add the use-statement

File: Helpers/ProjectHelper.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public function getShop()
+    {
+        try {
+            $shop = Shopware()->Shop();
+        } catch (Exception $e) {
+            $shop = null;
+        }
+
+        if (!($shop instanceof \Shopware\Models\Shop\Shop)) {
```
Comment (sobyte) [inline Helpers/ProjectHelper.php]: Add use-statement

File: Helpers/ProjectHelper.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            $shop = Shopware()->Models()
+                ->getRepository('Shopware\Models\Shop\Shop')
```
Comment (sobyte) [inline Helpers/ProjectHelper.php]: Please use `Shop::class`

File: Helpers/ProjectHelper.php (hunk 2 above)
Comment (sobyte) [reply Helpers/ProjectHelper.php]: Also `$this->modelManager`

File: Helpers/ProjectHelper.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    public function getShop()
+    {
+        try {
+            $shop = Shopware()->Shop();
+        } catch (Exception $e) {
+            $shop = null;
+        }
+
+        if (!($shop instanceof \Shopware\Models\Shop\Shop)) {
+            $shop = Shopware()->Models()
```
Comment (sobyte) [inline Helpers/ProjectHelper.php]: `$this->modelManager`

File: K10rProject.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * @param null|string $oldVersion
+     * @param null|string $newVersion
+     *
+     * @return bool
+     */
+    private function applyUpdates($oldVersion = null, $newVersion = null)
+    {
+        $versionClosures = [
+            '1.0.0' => function () {
```
Comment (sobyte) [inline K10rProject.php]: Shouldn't this be `2.0.0`?

File: README.md [hunk 7]
```diff
@@ -1,30 +1,38 @@
...
 ## Usage
 ### Installation
 __This is a project specific plugin, so make sure every project has it's own copy of this base plugin.__
-* Download the ZIP or clone this repository into your `engine/Shopware/Plugins/Local/Core/` folder.
+* Download the ZIP or clone this repository into your `custom/plugins/` folder.
 
 ### Create a Migration
-* Activate the plugin in your local development instance
-* Use the shopware command line to create a new migration: `php bin/console k10r:migration:create`
+* Install and activate `NetcomMigrations` in your local development instance
```
Comment (sobyte) [inline README.md]: Can you add an information on where to get the plugin?

File: composer.json [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
   "authors": [
     {
-      "name": "Siegmund Mioduszewski",
-      "email": "s.mioduszewski@k10r.de"
-    },
-    {
-      "name": "Uwe Kleinmann",
-      "email": "u.kleinmann@k10r.de"
+      "name": "Kellerkinder GmbH",
+      "email": "hallo@kellerkinder.de"
```
Comment (sobyte) [inline composer.json]: We use `anfragen@` for other plugins and in the plugin builder :)

**Review by sobyte — CHANGES_REQUESTED** [review]

File: Helpers/LicenseHelper.php (hunk 1 above)
Comment (kleinmann) [reply Helpers/LicenseHelper.php:30]: This functionality is not used in 5.5 anymore, as there's no code encryption/ionCube present. I can, however, output a deprecation error.

File: Helpers/ProjectHelper.php (hunk 2 above)
Comment (kleinmann) [reply Helpers/ProjectHelper.php]: Fixed :-)

File: Helpers/ProjectHelper.php (hunk 3 above)
Comment (kleinmann) [reply Helpers/ProjectHelper.php]: Fixed :-)

File: Helpers/ProjectHelper.php (hunk 4 above)
Comment (kleinmann) [reply Helpers/ProjectHelper.php]: Fixed :-)

File: Helpers/ProjectHelper.php (hunk 5 above)
Comment (kleinmann) [reply Helpers/ProjectHelper.php]: Fixed

File: K10rProject.php (hunk 6 above)
Comment (kleinmann) [reply K10rProject.php]: Yes :-) Fixed.

File: composer.json (hunk 8 above)
Comment (kleinmann) [reply composer.json]: Good catch, changed.

File: README.md (hunk 7 above)
Comment (kleinmann) [reply README.md]: Added links in the README :-)

File: Helpers/LicenseHelper.php (hunk 1 above)
Comment (sobyte) [reply Helpers/LicenseHelper.php:30]: Ah! Haven't thought about that 😊Deprecation message is ok

**Review by sobyte — APPROVED** [review]

**Pull request merged**